### DISCUSSION TITLE
Kafka Producer Metrics are only showing internal topics data

### DISCUSSION THREAD

**Discussion by srijita-mallick** [strimzi/strimzi-kafka-operator#8023] [Q&A] [answered]
We have enabled Kafka Exporter metrics for our cluster. 
When I query kafka topic metrics like `kafka_topic_partitions`, it shows the respective data for all topics.
But when I query kafka producer metrics like `kafka_producer_producer_topic_metrics_record_send_rate`, , it shows the data for only internal topics.
Below is the screenshot for same -
<img width="1664" alt="Screenshot 2023-02-02 at 9 11 56 AM" src="https://user-images.githubusercontent.com/122457427/216226175-39564960-d402-4a6f-9564-752b2a99f410.png">

Do I have to add something more in the config for exporter or jmx?

**Comment by scholzj** [comment, marked as answer]
Not sure I got you right. Strimzi and the Kafka Exporter or the JMX Prometheus agent in the Kafka operands monitor only them. They do not collect any metrics from your Kafka clients. So what exactly do you expect to see there? I guess what you see are the _producers_ which are part of the operands and these are obviously mostly using internal topics only. If you want to monitor your own producers or consumers, you need to configure monitoring for them.

**Reply by srijita-mallick** [reply]
Thank you for the great explanation. I looked into it [here](https://www.linkedin.com/pulse/kafka-monitor-producer-metrics-using-jmx-prometheus-hari-ramesh/) and realised i need to also expose the Kafka Producer Client to get those metrics. Thanks!
